AzureTable/Microsoft.DataTransfer.AzureTable/RemoteLogging/RemoteLogging.cs

```
@@ -0,0 +1,58 @@
+        private readonly string tableName = "migrationLogs";
+
+        /// <summary>
+        /// Remote logger constructor
+        /// </summary>
+        /// <param name="storageAccount"></param>
+        /// <param name="connectionPolicy"></param>
+        public RemoteLogging(CloudStorageAccount storageAccount, TableConnectionPolicy connectionPolicy)
+        {
+            CloudTableClient tcMigrationLogger = storageAccount.CreateCloudTableClient(connectionPolicy: connectionPolicy);
+            migrationLogger = tcMigrationLogger.GetTableReference(tableName);
+        }
+
+        /// <summary>
+        /// Create a Table if it does not exists already
+        /// </summary>
+        public async Task<bool> CreateRemoteLoggingTableIfNotExists(CancellationToken cancellation)
+        {
+            return await migrationLogger.CreateIfNotExistsAsync(IndexingMode.Consistent, tableThroughput, cancellation);
+        }
+        /// <summary>
+        /// Log the failures that occurred as a result of using DT. Given this is logging code, it will not throw errors to prevent
+        /// crashing the application
+        /// </summary>
+        /// <param name="partitionKey"></param>
+        /// <param name="rowKeys"></param>
+        /// <param name="exception"></param>
+        /// <param name="additionalDetails"></param>
+        public async void LogFailures(string partitionKey, string rowKeys, string exception, string additionalDetails = null)
+        {
+            try
+            {
+                //Log the failure to a cosmosDB table in  the provided account.
+                LoggingTableEntity log = new LoggingTableEntity(partitionKey, rowKeys,
+                    exception, Environment.MachineName, additionalDetails);
+                TableOperation loggingOp = TableOperation.InsertOrReplace(log);
+                TableResult result = await migrationLogger.ExecuteAsync(loggingOp);
+            }
+            catch { }
```

There are many other places where I believe exception can be thrown by this new logging code. For example, CreateRemoteLoggingTableIfNotExists. Do you want to ignore exceptions there as well?

---

agree. however, in the updated code, we want an exception if a table is not created as it is an intentional user choice, so not eating up the exception there.